src/wallet/sqlite.cpp

```diff
@@ -115,6 +115,43 @@ void SQLiteDatabase::Cleanup()
     }
 }
 
+bool SQLiteDatabase::Verify(bilingual_str& error)
+{
+    assert(m_db);
+
+    sqlite3_stmt* stmt{nullptr};
+    ret = sqlite3_prepare_v2(m_db, "PRAGMA integrity_check", -1, &stmt, nullptr);
+    if (ret != SQLITE_OK) {
+        sqlite3_finalize(stmt);
+        error = strprintf(_("SQLiteDatabase: Failed to prepare statement to verify database: %s"), sqlite3_errstr(ret));
+        return false;
+    }
+    while (true) {
+        ret = sqlite3_step(stmt);
+        if (ret == SQLITE_DONE) {
+            break;
+        } else if (ret != SQLITE_ROW) {
+            error = strprintf(_("SQLiteDatabase: Failed to execute statement to verify database: %s"), sqlite3_errstr(ret));
+            break;
+        }
+        const char* msg = (const char*)sqlite3_column_text(stmt, 0);
+        if (!msg) {
+            error = strprintf(_("SQLiteDatabase: Failed to read database verification error: %s"), sqlite3_errstr(ret));
+            break;
+        }
+        std::string str_msg(msg);
+        if (str_msg == "ok") {
+            continue;
+        }
+        if (error.original.empty()) {
```

Comment:
c3aeda3d2f7d7b34b8a35ccea478161b0ca2c80d
```suggestion
        if (error.empty()) {
```

Reply:
Done

Reply:
Not yet :)

Reply:
Oops.